modules/virtual_network_gateway/main.tf

```
@@ -2,78 +2,78 @@
+  name                = var.name
   tags                = var.tags
 
-  type     = try(var.type, "Vpn")
-  vpn_type = try(var.vpn_type, "RouteBased")
-  sku      = try(var.sku, "Basic")
+  type     = var.type
+  vpn_type = var.vpn_type
+  sku      = var.sku
 
-  active_active                    = try(var.active_active, false)
-  default_local_network_gateway_id = try(var.default_local_network_gateway_id, null)
-  edge_zone                        = try(var.edge_zone, null)
-  enable_bgp                       = try(var.enable_bgp, false)
-  generation                       = try(var.generation, null)
-  private_ip_address_enabled       = try(var.private_ip_address_enabled, null)
+  active_active                    = var.active_active
+  default_local_network_gateway_id = var.default_local_network_gateway_id
+  edge_zone                        = var.edge_zone
+  enable_bgp                       = var.enable_bgp
+  generation                       = var.generation
+  private_ip_address_enabled       = var.private_ip_address_enabled
 
   dynamic "ip_configuration" {
     for_each = var.ip_configuration
     content {
       name                          = ip_configuration.value.name
-      public_ip_address_id          = try(ip_configuration.value.create_public_ip, false) ? azurerm_public_ip.this[ip_configuration.value.name].id : try(data.azurerm_public_ip.exists[ip_configuration.value.name].id, null)
-      private_ip_address_allocation = try(ip_configuration.value.private_ip_address_allocation, "Dynamic")
+      public_ip_address_id          = ip_configuration.value.create_public_ip ? azurerm_public_ip.this[ip_configuration.value.name].id : data.azurerm_public_ip.exists[ip_configuration.value.name].id
+      private_ip_address_allocation = ip_configuration.value.private_ip_address_allocation
       subnet_id                     = ip_configuration.value.subnet_id
     }
   }
 
   dynamic "vpn_client_configuration" {
     for_each = var.vpn_client_configuration
     content {
-      address_space = try(vpn_client_configuration.value.address_space, null)
-      aad_tenant    = try(vpn_client_configuration.value.aad_tenant, null)
-      aad_audience  = try(vpn_client_configuration.value.aad_audience, null)
-      aad_issuer    = try(vpn_client_configuration.value.aad_issuer, null)
+      address_space = vpn_client_configuration.value.address_space
+      aad_tenant    = vpn_client_configuration.value.aad_tenant
+      aad_audience  = vpn_client_configuration.value.aad_audience
+      aad_issuer    = vpn_client_configuration.value.aad_issuer
       dynamic "root_certificate" {
-        for_each = try(vpn_client_configuration.value.root_certificate, null) != null ? { for t in vpn_client_configuration.value.root_certificate : t.name => t } : {}
+        for_each = vpn_client_configuration.value.root_certificate != null ? { for t in vpn_client_configuration.value.root_certificate : t.name => t } : {}
```

instead of doing a condition against `null` you can use coalesce, like this:
```suggestion
        for_each = { for t in coalesce(vpn_client_configuration.value.root_certificate, {}) : t.name => t }
```

---

thanks, changed